charmcraft.yaml

```diff
@@ -1,7 +1,70 @@
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
-
+name: tempo-k8s
 type: charm
+
+assumes:
+  - k8s-api
+
+  # Juju 3.0.3+ needed for secrets and open-port
+  - juju >= 3.0.3
+
+description: |
+  Tempo is a distributed tracing backend by Grafana.
+summary: |
+  Tempo is a distributed tracing backend by Grafana.
+
+containers:
+  tempo:
+    resource: tempo-image
+    mounts:
+      - storage: data
+        location: /tmp/tempo
+  nginx:
+    resource: nginx-image
+
+resources:
+  tempo-image:
+    type: oci-image
+    description: OCI image for Tempo
+    # Included for simplicity in integration tests
+    # see https://hub.docker.com/r/grafana/tempo/tags
+    upstream-source: grafana/tempo:2.4.0
+  nginx-image:
+    type: oci-image
+    description: OCI image for nginx
+    upstream-source: nginx:latest
```

Comment:
Idea: how about we keep it at `1.25` so that the version isn't floating too far? https://hub.docker.com/_/nginx

This would mean that we'll need to make sure we upgrade it later but at least we'd have some control over it.